hermes-management/src/main/java/pl/allegro/tech/hermes/management/domain/topic/TopicWithSchemaService.java

```diff
@@ -0,0 +1,99 @@
+package pl.allegro.tech.hermes.management.domain.topic;
+
+import org.springframework.stereotype.Component;
+import pl.allegro.tech.hermes.api.PatchData;
+import pl.allegro.tech.hermes.api.RawSchema;
+import pl.allegro.tech.hermes.api.Topic;
+import pl.allegro.tech.hermes.api.TopicName;
+import pl.allegro.tech.hermes.api.TopicWithSchema;
+import pl.allegro.tech.hermes.domain.topic.TopicAlreadyExistsException;
+import pl.allegro.tech.hermes.management.config.TopicProperties;
+import pl.allegro.tech.hermes.management.domain.topic.schema.SchemaService;
+import pl.allegro.tech.hermes.management.domain.topic.validator.TopicValidator;
+
+import java.util.Map;
+import java.util.Optional;
+
+import static pl.allegro.tech.hermes.api.ContentType.AVRO;
+
+@Component
+public class TopicWithSchemaService {
+
+    private final TopicService topicService;
+
+    private final TopicValidator topicValidator;
+
+    private final SchemaService schemaService;
+
+    private final TopicProperties topicProperties;
+
+    public TopicWithSchemaService(TopicService topicService,
+                                  TopicValidator topicValidator,
+                                  SchemaService schemaService,
+                                  TopicProperties topicProperties) {
+        this.topicService = topicService;
+        this.topicValidator = topicValidator;
+        this.schemaService = schemaService;
+        this.topicProperties = topicProperties;
+    }
+
+    public TopicWithSchema getTopicWithSchema(TopicName topicName) {
+        Topic topic = topicService.getTopicDetails(topicName);
+        Optional<RawSchema> schema = Optional.empty();
+        if (isAvroTopic(topic)) {
+            schema = schemaService.getSchema(topicName.qualifiedName());
+        }
+        return new TopicWithSchema(topic, schema.map(RawSchema::value).orElse(null));
+    }
+
+    public void createTopicWithSchema(TopicWithSchema topicWithSchema, String createdBy, CreatorRights isAllowedToManage) {
```

Comment:
i see no transaction here - if topic creation fails, schema will be created and subsequent tries to create topic will fail..

since it is not always possible to delete schema, i would rather:

* create topic
* create schema
* if schema creation fails - delete topic